msal/src/test/java/com/microsoft/identity/client/robolectric/tests/OperationParametersTest.java

```diff
@@ -0,0 +1,192 @@
+        AcquireTokenSilentOperationParameters operationParameters = OperationParametersAdapter.createAcquireTokenSilentOperationParameters(getAcquireTokenSilentParametersWithoutClaims(), getConfiguration(AAD_NONE_CONFIG_FILE), getCache());
+
+        Assert.assertEquals(false, operationParameters.getForceRefresh());
+    }
+
+    private ClaimsRequest getAccessTokenClaimsRequest(@NonNull String claimName, @NonNull String claimValue){
+        ClaimsRequest cp1ClaimsRequest = new ClaimsRequest();
+        RequestedClaimAdditionalInformation info = new RequestedClaimAdditionalInformation();
+        info.setValues(new ArrayList<Object>(Arrays.asList(claimValue)));
+        cp1ClaimsRequest.requestClaimInAccessToken(claimName, info);
+        return cp1ClaimsRequest;
+    }
+
+    private AcquireTokenParameters getAcquireTokenParametersWithClaims(){
+        AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
+                .withClaims(getAccessTokenClaimsRequest("device_id", ""))
+                .withScopes(new ArrayList<String>(Arrays.asList("User.Read")))
+                .startAuthorizationFromActivity(mActivity)
+                .build();
+
+        return parameters;
+    }
+
+    private AcquireTokenParameters getAcquireTokenParametersWithoutClaims(){
+        AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
+                .withScopes(new ArrayList<String>(Arrays.asList("User.Read")))
+                .startAuthorizationFromActivity(mActivity)
+                .build();
+
+        return parameters;
+    }
+
+    private AcquireTokenSilentParameters getAcquireTokenSilentParametersWithClaims() {
+        AcquireTokenSilentParameters parameters = new AcquireTokenSilentParameters.Builder()
+                .withClaims(getAccessTokenClaimsRequest("device_id", ""))
+                .withScopes(new ArrayList<String>(Arrays.asList("User.Read")))
+                .fromAuthority("https://login.microsoftonline.com/common")
+                .build();
+
+        return parameters;
+    }
+
+    private AcquireTokenSilentParameters getAcquireTokenSilentParametersWithoutClaims(){
+        AcquireTokenSilentParameters parameters = new AcquireTokenSilentParameters.Builder()
+                .withScopes(new ArrayList<String>(Arrays.asList("User.Read")))
+                .fromAuthority("https://login.microsoftonline.com/common")
+                .build();
+
+        return parameters;
+    }
+
+    private PublicClientApplicationConfiguration getConfiguration(String path){
+        return PublicClientApplicationConfigurationFactory.initializeConfiguration(mContext, getConfigFile(path));
+    }
+
+    private OAuth2TokenCache getCache(){
+        return new TestOAuthTokenCachen(mContext, null, null);
+    }
+
+    private File getConfigFile(String path){
+        return new File(path);
+    }
+
+    private class TestOAuthTokenCachen extends MsalOAuth2TokenCache {
```

Comment:
Duplicated code. This class was already created in another file above, it might be better to have a separate java source file for it and import it wherever needed.

Reply:
Also some comments as above about class name and javadoc